### ISSUE TITLE
Install fails with non-ASCII characters in Windows user path (install_github)

### ISSUE THREAD

**Issue by jogrue** [r-lib/remotes#114]
I am not able to install packages from GitHub with `devtools` (https://github.com/hadley/devtools/issues/1637), I now found out that using `remotes` returns the same error message and this is the right place to submit the bug report.

`install_github` fails for me because my Windows username contains an umlaut ("ü"). I would not set up my Windows account that way anymore, but still, this should probably be fixed. I found a similar (supposedly fixed) issue from 2015: https://github.com/hadley/devtools/issues/676

The error happens for every package. It fails when my user path `C:/Users/Johann Gründl/` is read as `C:/Users/Johann Gr|ndl` for `DESCRIPTION`:
```
* installing *source* package 'httr' ...
Warning in file(file, if (append) "a" else "w") :
  cannot open file 'C:/Users/Johann Gr|ndl/Documents/R/win-library/3.4/httr/DESCRIPTION': Invalid argument
Error in file(file, if (append) "a" else "w") : 
  cannot open the connection
ERROR: installing package DESCRIPTION failed for package 'httr'
* removing 'C:/Users/Johann Gründl/Documents/R/win-library/3.4/httr'
```
R version 3.4.2
remotes version 1.1.0

I uploaded the full output of `install_github `and `sessionInfo()` here:
https://gist.github.com/jogrue/c55a95fe9890f83d56acf96b353c9631

**Comment by jogrue** [comment]
I now had the same issue with utils::install.packages where the type is *source*. So it might not be an devtools/remotes issue. I tried to find a workaround, but did not succeed. I tried to use shortPathName() versions of my lib paths in the environment variables, but this did not help. Maybe it is related to .libPaths() using normalizePath (which returned my shortPathName versions to the long versions). Maybe just using a simple personal library like C:\rlib would help. I just renamed my user folder (http://www.opentechguides.com/how-to/article/windows-10/43/win10-change-account.html).

I still believe that this should be fixed, but it is probably not an issue with devtools/remotes? Unfortunately, I am not able (or willing to be honest) to investigate this further because I usually run R on Linux anyway and I was able to fix it by changing my Windows user folder.
